### PR TITLE
fix(quantic): Fix some styling issues for the source citations component

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5134](https://coveord.atlassian.net/browse/SFINT-5134)

Following discussions with @adroletCoveo : 

- Updated the height of the citations badges to be 32px instead of 30px
- Removed the capitalized 'M' in the label 'Learn more'
- Increased the font size of the title and label of the citations to 14px from 13px
- I can also confirm that the title color is not blue when not on hover/focus, but changed the color to match the learn more label and be a bit more gray

BEFORE:

Screenshot 2023-08-09 at 9 38 39 AM

AFTER:

Screenshot 2023-08-09 at 9 32 43 AM

[SFINT-5134]: https://coveord.atlassian.net/browse/SFINT-5134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mmitiche** [comment]
A before/After screenshots are usually very useful when adding such styling changes.

**Review by mmitiche — APPROVED** [review]
I would wait for the approval of the UX team to confirm that we didn't forget anything but it's a yes from me ✅

**Review by adroletCoveo — APPROVED** [review]
Thank you Simon, this is perfect!

The hover and focus effects seemed fine in the video too.

**Pull request merged**
